.github/python_actions/genContractLists.py

```diff
@@ -18,28 +19,36 @@
 
 
 
-def address_directory():
-   r = requests.get(ADDRESSBOOK_URL)
-   return r.json()
+def address_directory(chain, status=None):
+    r = requests.get(f"https://raw.githubusercontent.com/balancer/balancer-deployments/master/addresses/{chain}.json")
+    r=r.json()
+    if isinstance(status, str):
+        filtered = {}
+        for deployment, depdata in r.items():
+            if depdata["status"] == status:
+                filtered[deployment] = depdata
+        return filtered
```

Comment:
```suggestion
        return {k: v for k, v in r.items() if v['status'] == status}
```

You can use dict comprehension or `filter` for a one-liner.